### PR TITLE
[RSDK-6312]   Change OAK module method return types to ViamImage

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-6312](https://viam.atlassian.net/browse/RSDK-6312)

Update our module's Viam Python SDK dependency to consume ViamImage. Refactors and code cleanup added as appropriate

Changes
- Created `helpers.py` to take care of shared classes and image transcoding helpers
- Moved encode depth raw func from OAK method to `helpers.py` as a standalone func
- Added encode jpeg bytes as shared logic rather than spaghetti
- Changed `get_image` signature to return ViamImage
- Updated requirements deps list to account for numpy breaking change and Viam SDK changes

[RSDK-6312]: https://viam.atlassian.net/browse/RSDK-6312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### Color only (get_image)
```
{
  "sensors": [
    "color"
  ]
}
``` 
Color stream looks good on app

### Depth only (get_image)
```
{
  "sensors": [
    "depth"
  ]
}
```
Depth stream looks good on app

### Color and depth (get_images)
```
{
  "sensors": [
    "color",
    "depth"
  ]
}
```
Tested with my python-sdk client:
```
async def connect():
    opts = RobotClient.Options.with_api_key(
      api_key='<secret>',
      api_key_id='<secret-id>'
    )
    address = '<addy>'

    return await RobotClient.at_address(address, opts)

async def get_images(cam: Camera):
    images = await cam.get_images()
    assert len(images) == 2 
    assert len(images[0]) == 2
    color_img, depth_map = images[0][0], images[0][1]
    metadata = images[1]
    print(f"captured at: {metadata.captured_at}")
    check_jpeg_image_from_get_images(color_img)
    check_depth_map_from_get_images(depth_map)
   

def check_jpeg_image_from_get_images(named_image: NamedImage):
    image = named_image.image
    if image is None:
        raise Exception("image is None")
    np_array = np.array(image)
    print("color image format: " + image.format)
    print("color image shape: " + str(np_array.shape))

def check_depth_map_from_get_images(depth_map: NamedImage):
    print(f'{len(depth_map.data)}')
    depth_arr = depth_map.bytes_to_depth_array()
    np_array = np.asarray(depth_arr, dtype=np.uint16)
    print("depth array shape: " + str(np_array.shape))

"""
The main function is the one called upon script execution. It sets up and calls the
two other functions.
"""
async def main():
    robot = await connect()
    cam = Camera.from_robot(robot, 'oak-cam')

    await get_images(cam)
    await cam.close()
    await robot.close()


if __name__ == '__main__':
  asyncio.run(main())
```
Outputted:
```
captured at: seconds: 165
nanos: 173421999

color image format: JPEG
color image shape: (400, 640, 3)
512024
depth array shape: (400, 640)
```
The key part is that `bytes_to_depth_array` was successful. Meaning there is no format error for the depth map with the new ViamImage under-the-hood impl

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 depthai-sdk==1.13.1
-viam-sdk==0.11.0
+viam-sdk==0.25.2
+Pillow==10.4.0
```
Comment (hexbabe) [inline requirements-dev.txt:3]: Pillow was removed as an implicit dep in our Python SDK https://github.com/viamrobotics/viam-python-sdk/pull/567

File: src/helpers.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        is_depth (bool): whether arr is depth data or RGB data
+
+    Returns:
+        bytes: JPEG bytes
+    """
+    if is_depth:
+        pil_image = Image.fromarray(arr, "I;16").convert("RGB")
+    else:
+        pil_image = Image.fromarray(arr)
+    output_buffer = io.BytesIO()
```
Comment (seanavery) [inline src/helpers.py:151]: [possible nit] Should we use `with` here to to ensure resource is closed out.

```
        with io.BytesIO() as output_buffer:
            pil_image.save(output_buffer, format="JPEG")
            raw_bytes = output_buffer.getvalue()

**Review by seanavery — APPROVED** [review]
LGTM

File: src/helpers.py (hunk 2 above)
Comment (hexbabe) [reply src/helpers.py]: Done

**Pull request merged**